### DISCUSSION TITLE
Annotated[type[T]] cannot be assigned to type[T]

### DISCUSSION THREAD

**Discussion by RobertCraigie** [microsoft/pyright#7421] [Q&A] [answered]
not sure if it's a bug or not but this code results in errors for me
```py
from __future__ import annotations

from typing import TypeVar
from typing_extensions import Annotated

class MyBase:
    ...

class MyClass(MyBase):
    ...

_T = TypeVar("_T", bound=MyBase)

def foo(type_: type[_T]) -> _T:
    ...

foo(MyClass)  # fine
foo(Annotated[MyClass, "metadata"])  # error
```
```
error: Argument of type "type[MyClass]" cannot be assigned to parameter "type_" of type "type[_T@foo]" in function "foo" (reportArgumentType)
```

I'm not fully aware of the semantics of `Annotated` but it would be quite nice if this worked and the return type was inferred to `MyClass`.

**Comment by erictraut** [comment, marked as answer]
This isn't an error. You're attempting to use `typing.Annotated` in a value expression. `Annotated` was designed to be used in type expressions (annotations), not in regular value expressions. The official typing spec recently clarified that when it is used in a value expression, it acts as a regular class, not as a special form — and that it is not compatible with `type[T]`. So pyright is correct here.

**Reply by RobertCraigie** [reply]
ah makes sense, thanks for the context!
